src/Workspaces/Core/Portable/DesignerAttribute/IDesignerAttributeServiceCallback.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.DesignerAttribute
+{
+    /// <summary>
+    /// Callback the host (VS) passes to the OOP service to allow it to send batch notifications
+    /// about designer attribute info.  There is no guarantee that the host will have done anything
+    /// with this data when the callback returns, only that it will try to inform the project system
+    /// about the designer attribute info in the future.
+    /// </summary>
+    internal interface IDesignerAttributeServiceCallback
```

Comment:
>IDesignerAttributeServiceCallback [](start = 23, length = 33)

Perhaps `IRemoteDesignerAttributeServiceCallback` would be better?

Reply:
i was hesitant about this.  i like that 'remote' means "implementation is on the remote side".  i.e. it's osmething you call when you're not remote and you want to talk to remote.  this is the reverse.  it's what the remote side will call into with results.  so it's *not* remote.  Perhaps IHostDesignerAttributeServiceCallback?

Reply:
I was thinking that we have `IRemoteDesignerAttributeService` that's running in OOP. And it is its callback to devenv, so IRemoteDesignerAttributeServiceCallback makes sense as a callback from the remote service.

---
In reply to: [392403071](https://github.com/dotnet/roslyn/pull/42346#discussion_r392403071) [](ancestors = 392403071)

Reply:
Alternatively, don't call it a callback. Call it IProjectSystemDesignerAttributeListener or something like that.

---
In reply to: [392404529](https://github.com/dotnet/roslyn/pull/42346#discussion_r392404529) [](ancestors = 392404529,392403071)

Reply:
sure.

Reply:
These are difficult to name as there are two parts to the callback, the little stub in the remote side, and the real impl that the stub calls into.  the place where most of the work happens is what i'm trying to have this interface name, and i'm trying to use 'remote' to indicate that you're making a call into OOP.  that fits the naming we've used elsewhere.

Reply:
Renamed to IDesignerAttributeListener